### PR TITLE
chore: AI Image Generator cleanup

### PR DESCRIPTION
**Pull request by mgoudy91**
## Purpose

- replaces React scripts with vite
- many required changes to make ^ work along side it

## Deployment

image

image

### CODE REVIEW

File: apps/ai-image-generator/frontend/index.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Vite + TS</title>
+  </head>
+  <body>
+    <div id="root"></div>
+    <script type="module" src="/src/index.tsx"></script>
```
Comment (mgoudy91) [inline apps/ai-image-generator/frontend/index.html:11]: Missing this script tag was a critical problem 🤦

File: apps/ai-image-generator/frontend/src/App.tsx [hunk 2]
```diff
@@ -2,21 +2,9 @@ import { locations } from '@contentful/app-sdk';
...
-import Home from './locations/Home';
 
 const ComponentLocationSettings = {
   [locations.LOCATION_APP_CONFIG]: ConfigScreen,
-  [locations.LOCATION_ENTRY_FIELD]: Field,
-  [locations.LOCATION_ENTRY_EDITOR]: EntryEditor,
-  [locations.LOCATION_DIALOG]: Dialog,
-  [locations.LOCATION_ENTRY_SIDEBAR]: Sidebar,
-  [locations.LOCATION_PAGE]: Page,
-  [locations.LOCATION_HOME]: Home,
```
Comment (mgoudy91) [inline apps/ai-image-generator/frontend/src/App.tsx:19]: these locations never meaningfully existed

File: apps/ai-image-generator/frontend/src/locations/ConfigScreen.spec.tsx
Comment (mgoudy91) [inline apps/ai-image-generator/frontend/src/locations/ConfigScreen.spec.tsx:1]: most of these location tests were just testing the default location components, which aren't used by this app

File: apps/ai-image-generator/package.json [hunk 3]
```diff
@@ -19,8 +19,11 @@
   "dependencies": {
     "@contentful/app-scripts": "1.13.1",
     "esbuild": "^0.23.0",
-    "sharp": "^0.33.0",
     "node-addon-api": "^7.0.0",
-    "node-gyp": "^10.0.1"
+    "node-gyp": "^10.0.1",
```
Comment (mgoudy91) [inline apps/ai-image-generator/package.json:23]: these were env fixes required for the general build to work. I'm guessing react-scripts included `lru-cache` in its tree somehow

File: apps/ai-image-generator/frontend/index.html (hunk 1 above)
Comment (sarahlessner) [reply apps/ai-image-generator/frontend/index.html:11]: Oh yeah I ran into that too on the first one I did.

File: apps/ai-image-generator/frontend/src/components/config/DisclaimerSection/DisclaimerSection.spec.tsx [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
 import { render, screen } from '@testing-library/react';
-import configPageCopies from 'constants/configPageCopies';
+import configPageCopies from '@constants/configPageCopies';
 
 const { getByText } = screen;
 const { sectionTitle, linkSubstring } = configPageCopies.disclaimerSection;
 
 describe('DisclaimerSection component', () => {
-  it('Component mounts without correct content', async () => {
+  it('Component mounts without correct coqntent', async () => {
```
Comment (sarahlessner) [inline apps/ai-image-generator/frontend/src/components/config/DisclaimerSection/DisclaimerSection.spec.tsx:9]: ?

**Review by sarahlessner — APPROVED** [review]
lgtm 👍

File: apps/ai-image-generator/frontend/src/components/config/DisclaimerSection/DisclaimerSection.spec.tsx (hunk 4 above)
Comment (mgoudy91) [reply apps/ai-image-generator/frontend/src/components/config/DisclaimerSection/DisclaimerSection.spec.tsx:9]: haha, great 👀 . Very unintentional

**Review by sarahlessner — APPROVED** [review]

**Pull request merged**